### PR TITLE
Add glossary definitions of packaging and metadata terms

### PR DESCRIPTION
**Pull request by befeleme**
The definitions were originally included in (the draft of) PEP 639, refining and extending the existing glossary terms.
Having the unified vocabulary in place is essential to accurately describe the complex Python packaging landscape.

----
📚 Documentation preview 📚: https://python-packaging-user-guide--1541.org.readthedocs.build/en/1541/glossary/#term-Built-Metadata

### CODE REVIEW

File: source/glossary.rst [hunk 1]
```diff
@@ -264,10 +357,22 @@ Glossary
...
+        originally introduced in :pep:`427`
+        and defined by the :ref:`binary-distribution-format` specification.
         See :ref:`package-formats` for more information.
+        Not to be confused with its reference implementation,
+        the :term:`Wheel Project`.
+
+
+    Wheel Project
+
+        The PyPA reference implementation of the :term:`Wheel Format`.
```
[the remark this answers is not in the document]
Comment (webknjaz) [reply source/glossary.rst:374]: This looks like an opportunity to make use of the `.. seealso::` directive. WDYT?

**Review by webknjaz — APPROVED** [review]
Looks good, I'll leave merging to @chrysle.

File: source/glossary.rst (hunk 1 above)
[the remark this answers is not in the document]
Comment (befeleme) [reply source/glossary.rst:374]: There are quite a few references to "see also" in the glossary - I'd either do it for all of them or none. I find the directive rendering quite disruptive when looking at the page, but I don't have a strong opinion here.

Please compare:
image
vs
image

[the remark this answers is not in the document]
Comment (webknjaz) [reply source/glossary.rst]: Okay, fair. Marking as resolved.

**Pull request merged**